src/test/libcephfs/snapsize.cc

```diff
@@ -0,0 +1,219 @@
+  std::cout << " mksnap 'snapA'" << std::endl;
+  // snapdir '.snap/' size
+  ASSERT_EQ(ceph_statx(cmount, c_snapdir_path, &stx, CEPH_STATX_SIZE, 0), 0);
+  std::cout << " path " << c_snapdir_path << " size " << stx.stx_size << std::endl;
+  ASSERT_EQ(stx.stx_size, 3);
+  // snapshot 'snapA' size
+  sprintf(c_snapshot_path, "%s/snapA", c_snapdir_path);
+  ASSERT_EQ(ceph_statx(cmount, c_snapshot_path, &stx, CEPH_STATX_SIZE, 0), 0);
+  std::cout << " path " << c_snapshot_path << " size " << stx.stx_size << std::endl;
+  ASSERT_EQ(stx.stx_size, 3);
+
+  // ****** Test 'snapB' ******
+  write_file(file_path, "defg");
+  ASSERT_EQ(0, ceph_mksnap(cmount, c_snapdir_path, "snapB", 0755, nullptr, 0));
+  std::cout << " mksnap 'snapB'" << std::endl;
+  // snapdir '.snap/' size
+  ASSERT_EQ(ceph_statx(cmount, c_snapdir_path, &stx, CEPH_STATX_SIZE, 0), 0);
+  std::cout << " path " << c_snapdir_path << " size " << stx.stx_size << std::endl;
+  ASSERT_EQ(stx.stx_size, 7);
+  // snapshot 'snapB' size
+  sprintf(c_snapshot_path, "%s/snapB", c_snapdir_path);
+  ASSERT_EQ(ceph_statx(cmount, c_snapshot_path, &stx, CEPH_STATX_SIZE, 0), 0);
+  std::cout << " path " << c_snapshot_path << " size " << stx.stx_size << std::endl;
+  ASSERT_EQ(stx.stx_size, 4);
+
+  // Write to the file and test the snapdir '.snap/' and the size shouldn't
+  // change
+  write_file(file_path, "hijkl");
+  // snapdir '.snap/' size
+  ASSERT_EQ(ceph_statx(cmount, c_snapdir_path, &stx, CEPH_STATX_SIZE, 0), 0);
+  std::cout << " path " << c_snapdir_path << " size " << stx.stx_size << std::endl;
+  ASSERT_EQ(stx.stx_size, 7);
+
+  // ****** Test 'snapC' ******
+  ASSERT_EQ(0, ceph_mksnap(cmount, c_snapdir_path, "snapC", 0755, nullptr, 0));
+  std::cout << " mksnap 'snapC'" << std::endl;
+  // snapdir '.snap/' size
+  ASSERT_EQ(ceph_statx(cmount, c_snapdir_path, &stx, CEPH_STATX_SIZE, 0), 0);
+  std::cout << " path " << c_snapdir_path << " size " << stx.stx_size << std::endl;
+  ASSERT_EQ(stx.stx_size, 12);
+  // snapshot 'snapC' size
+  sprintf(c_snapshot_path, "%s/snapC", c_snapdir_path);
+  ASSERT_EQ(ceph_statx(cmount, c_snapshot_path, &stx, CEPH_STATX_SIZE, 0), 0);
+  std::cout << " path " << c_snapshot_path << " size " << stx.stx_size << std::endl;
+  ASSERT_EQ(stx.stx_size, 5);
+
+  // ****** Test removing 'snapA' ******
+  ASSERT_EQ(0, ceph_rmsnap(cmount, c_snapdir_path, "snapA"));
+  std::cout << " rmsnap 'snapC'" << std::endl;
+  // snapdir '.snap/' size
+  ASSERT_EQ(ceph_statx(cmount, c_snapdir_path, &stx, CEPH_STATX_SIZE, 0), 0);
+  std::cout << " path " << c_snapdir_path << " size " << stx.stx_size << std::endl;
+  ASSERT_EQ(stx.stx_size, 9);
+
+  ASSERT_EQ(0, ceph_rmsnap(cmount, c_snapdir_path, "snapB"));
+  ASSERT_EQ(0, ceph_rmsnap(cmount, c_snapdir_path, "snapC"));
+  ceph_unlink(cmount, file_path);
+  ASSERT_EQ(ceph_rmdir(cmount, dir_path), 0);
+  ceph_shutdown(cmount);
+}
+
+TEST(SnapSize, NonRstats) {
```

Comment:
This test failed during QA run -
`2024-04-20T21:28:44.473 INFO:tasks.workunit.client.0.smithi136.stdout:[  FAILED  ] SnapSize.NonRstats`

Reply:
Okay, I will check it. This PR has been for a long time and maybe  the upstream has changed a lot.

Reply:
Let me know when changes have been made and it's ready for QA.